pkg/processor/procbuiltin/unwraprecord.go

```diff
@@ -0,0 +1,185 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procbuiltin
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/multierror"
+	"github.com/conduitio/conduit/pkg/processor"
+	"github.com/conduitio/conduit/pkg/record"
+)
+
+const (
+	unwrapRecordPayloadProcType = "unwraprecord"
+
+	// todo: support openCDC format
+	unwrapRecordConfigFormat = "format" // only debezium for now
+
+	RecordFormatDebezium     = "debezium"
+	RecordFormatKafkaConnect = "kafka-connect"
+)
+
+func init() {
+	processor.GlobalBuilderRegistry.MustRegister(unwrapRecordPayloadProcType, UnwrapRecordPayload)
+}
+
+// UnwrapRecordPayload unwraps the record format from the payload and returns the record as openCDC format
+func UnwrapRecordPayload(config processor.Config) (processor.Interface, error) {
+	return unwrapRecord(unwrapRecordPayloadProcType, recordPayloadGetSetter{}, config)
+}
+
+func unwrapRecord(
+	processorType string,
+	getSetter recordDataGetSetter,
+	config processor.Config,
+) (processor.Interface, error) {
+	var (
+		err    error
+		format string
+		r      record.Record
+	)
+
+	format = config.Settings[unwrapRecordConfigFormat]
+	if format == "" {
+		return nil, cerrors.Errorf("%s: record format type not specified", processorType)
+	}
+	if format != RecordFormatDebezium && format != RecordFormatKafkaConnect {
+		return nil, cerrors.Errorf("%s: record format should be one of {%s, %s}", processorType, RecordFormatDebezium, RecordFormatKafkaConnect)
+	}
+
+	return processor.InterfaceFunc(func(_ context.Context, rec record.Record) (record.Record, error) {
+		data := getSetter.Get(rec) // record.payload.after
```

Comment:
What does the comment indicate?

Reply:
that we're getting `record.payload.after` from this method, but wasn't that helpful honestly.
either way, it's gone with the refactor 👀